src/emc/nml_intf/interpl.cc

```diff
@@ -85,8 +85,8 @@ int NML_INTERP_LIST::append(NMLmsg * nml_msg_ptr)
 #ifdef DEBUG_INTERPL
     if (sizeof(temp_node) < MAX_NML_COMMAND_SIZE + 4 ||
 	sizeof(temp_node) > MAX_NML_COMMAND_SIZE + 16 ||
-	((void *) &temp_node.line_number) >
-	((void *) &temp_node.command.commandbuf)) {
+	((unsigned long) (void *) &temp_node.line_number) >
+	((unsigned long) (void *) &temp_node.command.commandbuf)) {
```

Comment:
From the language perspective, you cannot safely cast from `void *` to `unsigned long` as there is no guarantee that `unsigned long` will be capable of holding a value of pointer.

If you really want to cast `void *` to `integer` type please consider using `uintptr_t` type.